executor/containerdexecutor/executor.go

```diff
@@ -329,5 +296,73 @@ func (w *containerdExecutor) Exec(ctx context.Context, id string, process execut
 	if err != nil {
 		return errors.WithStack(err)
 	}
-	return taskProcess.Start(ctx)
+
+	err = w.runProcess(ctx, taskProcess, process.Resize, nil)
+	return err
+}
+
+func (w *containerdExecutor) runProcess(ctx context.Context, p containerd.Process, resize <-chan executor.WinSize, started func()) error {
+	statusCh, err := p.Wait(context.Background())
```

Comment:
why `context.Background`? Maybe leave a comment for future readers as well.

Reply:
Good question, I was wondering the same thing, this code was just moved from `Run` to share with `Exec`.   We need to ask your 3yr younger self :)
https://github.com/moby/buildkit/blame/master/executor/containerdexecutor/executor.go#L213

My hunch is that we wanted to keep the statusCh alive if we get ctx.Done so that we can send sigkill and capture status from that?

Reply:
Looking at `Wait` implementation I now understand it. `Wait` is non-blocking irrelevant from the context passed and context only affects the `statusCh` that we shouldn't cancel.

Reply:
I added a comment with your findings.